File: Aparna/Notepad/Saving/SaveAsPDF.cs
using iTextSharp.text;
using iTextSharp.text.pdf;
using System;
using System.IO;


namespace Notepad.Saving
{
    public class SaveAsPDF : SaveToSystem
    {
        public SaveAsPDF(string fileName,string text) : base(fileName, text) { }   
        public override void Save()
        {
            try
            {
                if (!String.IsNullOrEmpty(Text))
                {
                    Document document = new Document();
                    PdfWriter.GetInstance(document, new FileStream(FileName, FileMode.Create));
                    document.Open();
                    document.Add(new Paragraph(Text));
                    document.Close();
                }
            }
            catch
            {
                throw;
            }
        }
    }
}

File: Jayasree/Final Notepad 1.0/src/com/example/notepad/actionlistener/SaveAsPDF.java
package com.example.notepad.actionlistener;

import java.io.FileNotFoundException;
import java.io.FileOutputStream;

import javax.swing.JFileChooser;
import javax.swing.JMenuItem;
import javax.swing.JTextArea;

import com.itextpdf.text.Document;
import com.itextpdf.text.DocumentException;
import com.itextpdf.text.Paragraph;
import com.itextpdf.text.pdf.PdfWriter;

public class SaveAsPDF {
	public void saveaspdf(JMenuItem save, JTextArea area){
		JFileChooser fileChooser = new JFileChooser();
		fileChooser.setDialogTitle("Specify a file to save");

		int userSelection = fileChooser.showSaveDialog(save);
		if (userSelection == JFileChooser.APPROVE_OPTION) {
			saveasTextToPDF(fileChooser.getSelectedFile().toString(),area);
		}
	}
	private void saveasTextToPDF(String desc, JTextArea area) {
    	Document doc=new Document();
    	  try {
			PdfWriter.getInstance(doc, new FileOutputStream(desc));
    	  } catch (FileNotFoundException e) {
			e.printStackTrace();
    	  } catch (DocumentException e) {
			e.printStackTrace();
		  }
    	  doc.open();
    	  for (String line : area.getText().split("\\n")){
     		try {
				doc.add(new Paragraph(line));
			} catch (DocumentException e) {
				e.printStackTrace();
			}
    	  }
    	  doc.close();
		
	}
}
